src/module.c

```diff
@@ -6461,6 +6461,38 @@ const char *RM_CallReplyProto(RedisModuleCallReply *reply, size_t *len) {
     return callReplyGetProto(reply, len);
 }
 
+/* Redis only exposes "side effects" of command execution at the end of an execution unit.
+ *
+ * Side effects include propagating replication/AOF operations and executing the commands of clients blocked on keys
+ * modified by commands
+ *
+ * In general, for modules, execution unit start and end with the taking/release of the GIL.  For module code executed
+ * on the Redis main thread, this is done implicitly by Redis itself, while for module code executed on a separate
+ * thread, this is done explicitly by the module itself.
+ *
+ * This function enables modules to flush the execution unit, thereby having side effects become visible during
+ * execution.
+ *
+ * Examples of usage might be that one is running multiple commands, one after another, with RM_Call and wants
+ * processCommand() like semantics, where unblocked clients will be executed before the next RM_Call is made.
+ *
+ * Note: while one Modules that calls another module via RM_Call won't nest if they are operating on a separate thread,
+ * as each will need to grab the GIL independently, they can nest if both operate on the main thread.  In that case,
+ * it doesn't make sense for the further nested module to be able to cause side effects to the top level module, and
+ * therefore this function only succeeds if running at an execution_nesting level of 1.
+ */
+int RM_FlushExecutionUnit() {
```

Comment:
IIRC, previous comments suggested that we add both a context, and a flags arguments, so that in the future we're able to extend this API without breaking it.